.github/workflows/require-encapsulation.yml

```diff
@@ -0,0 +1,21 @@
+name: Check require encapsulation
+
+on:
+  push:
+    branches:
+      - "**"
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ master ]
+
+jobs:
+  check-require-encapsulation:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '3.2'
+          bundler-cache: true # runs 'bundle install' and caches installed gems automatically
+      - run: rm -f .rspec
+      - run: bundle exec rspec spec/require_encapsulation_check.rb
```

Comment:
What's the rationale for a separate job?

If the file is named correctly it can be part of the usual `rake spec` tasks and tested across the matrix.